Whiskey/Functions/Resolve-WhiskeyTaskPathParameter.ps1

```diff
@@ -0,0 +1,143 @@
+function Resolve-WhiskeyTaskPathParameter
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        # An object that holds context about the current build and executing task.
+        [object]$TaskContext,
+
+        [Parameter(ValueFromPipeline=$true)]
+        [string]$Path,
+
+        [Parameter(Mandatory)]
+        [string]$PropertyName,
+
+        [Parameter(Mandatory)]
+        [Management.Automation.ParameterMetadata]$CmdParameter,
+
+        [Parameter(Mandatory)]
+        [Whiskey.Tasks.ValidatePathAttribute]$ValidateAsPathAttribute
+    )
+
+    begin
+    {
+        Set-StrictMode -Version 'Latest'
+
+        $pathIdx = -1
+    }
+
+    process
+    {
+        Set-StrictMode -Version 'Latest'
+
+        $pathIdx++
+
+        $result = $Path
+
+        if( -not $Path )
+        {
+            if( $ValidateAsPathAttribute.Mandatory )
+            {
+                Stop-WhiskeyTask -TaskContext $Context `
+                                 -PropertyName $CmdParameter.Name `
+                                 -Message ('{0} is mandatory.' -f $TaskProperty[$propertyName])

```

Comment:
Where is `$TaskProperty` getting defined?

Reply:
I believe that was copied over from `Get-TaskParameter`. I changed it to just `$PropertyName` since that is a parameter inside `Resolve-WhiskeyTaskPathParameter`. Dumb mistake on my part.

Reply:
This should have failed a test. Please write a test that fails.

Reply:
You got it.